src/LIBSVM.jl

```diff
@@ -223,6 +228,36 @@ function check_train_input(X, y, kernel)
     end
 end
 
+function data2gram(kernel_function, X)
+    ntrain = size(X, 2)
+    gram = Array{Float64}(undef, ntrain, ntrain)
+    for i=1:ntrain
+        for j in 1:i
+            if i==j
+                gram[i,i] = kernel_function(X[:,i], X[:,i])
+            else
+                gram[i, j] = gram[j, i] = kernel_function(X[:, i], X[:, j])
+            end
```

Comment:
Doubling assignment on `i = j` is fine for me.
```suggestion
                gram[i, j] = gram[j, i] = kernel_function(X[:, i], X[:, j])
```